src/main/java/seedu/address/model/income/Income.java

```diff
@@ -0,0 +1,130 @@
+package seedu.address.model.income;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.address.model.claim.Amount;
+import seedu.address.model.claim.Description;
+import seedu.address.model.commonvariables.Name;
+import seedu.address.model.commonvariables.Phone;
+import seedu.address.model.tag.Tag;
+
+
+
+
+/**
+ * Represents an Income in the Financial Planner.
+ */
+public class Income {
+
+    // Identity fields
+    private final Name name;
+    private final Phone phone;
+    private final Description description;
+
+    //Data fields
+    private final Amount amount;
+    private final Set<Tag> tags = new HashSet<>();
+    private final Date dateAdded;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Income(Name name, Phone phone, Description description, Amount amount, Set<Tag> tags) {
+        requireAllNonNull(name, phone, description, amount, tags);
+        this.name = name;
+        this.phone = phone;
+        this.description = description;
+        this.amount = amount;
+        this.tags.addAll(tags);
+        this.dateAdded = new Date();
+    }
+
+    public Description getDescription() {
+        return description;
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Phone getPhone() {
+        return phone;
+    }
+
+    public Amount getAmount() {
+        return amount;
+    }
+
+    public Date getDateAdded() { return dateAdded; }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     * This defines a weaker notion of equality between two claims.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    /**
+     * Returns true if both incomes of the same name have at least one other identity field that is the same.
+     * This defines a weaker notion of equality between two incomes.
+     */
+    public boolean isSameIncome(Income otherIncome) {
+        if (otherIncome == this) {
+            return true;
+        }
+
+        return otherIncome != null
+                && otherIncome.getName().equals(getName())
+                && (otherIncome.getDescription().equals(getDescription()) || otherIncome.getPhone().equals(getPhone()));
```

Comment:
this line might be a little too long, can split into various variables possible?